Source/UINavigation/Private/UINavWidget.cpp

```diff
@@ -717,9 +719,39 @@ void UUINavWidget::NativeTick(const FGeometry & MyGeometry, float DeltaTime)
 {
 	Super::NativeTick(MyGeometry, DeltaTime);
 
-	if (IsSelectorValid() && bMovingSelector)
+	if (bUINavSetup)
```

Comment:
I feel like this bUINavSetup property is redundant. Can't you do something like this? If you could, that would remove the need for quite a bit of potentially redundant logic and properties.

`if (IsSelectorValid())
{
  if (bCompletedSetup)
  {
    ...
  }
  else
  {
    ...
    UINavSetup
    ...
  }`

Reply:
I reordered the code on 1546038, bUINavSetup is needed to call the function on next frame. Looking at the code this may happen more than once if rebuild navigation is called.